src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -43,8 +50,28 @@ void CommissionerDiscoveryController::ResetState()
     mReady              = true;
 }
 
+void CommissionerDiscoveryController::CheckValidSession()
+{
+    if (mReady)
+    {
+        return;
+    }
+    if (mUdcServer != nullptr)
+    {
+        UDCClientState * client = mUdcServer->GetUDCClients().FindUDCClientState(mCurrentInstance);
+        if (client != nullptr)
+        {
+            // everything looks good
+            return;
+        }
+    }
+    ResetState();
```

Comment:
This sounds like a side effect, but CheckValidSession() kind of sounds like a side-effect-free thing that should return a boolean.

What is this function actually doing?  It feels mis-named.